syntax = "proto3";

package quite.proto;

service ObjectService {
    rpc FindObject(ObjectRequest) returns (ObjectReply) {}
    rpc GetObjectProperty(PropertyRequest) returns (PropertyResponse) {}

    rpc MouseClick(MouseClickRequest) returns (VoidResponse) {}
}

message ObjectRequest {
  ObjectRequest parent = 1;
  string object_name = 2;
}

message ObjectReply {
  uint64 id = 1; // the internal id assigned by the probe
  string type_name = 2;
}

message PropertyRequest {
  uint64 id = 1;
  string property_name = 2;
}

message PropertyResponse {
  string type_name = 1;
  string value = 2;
}

message MouseClickRequest {
  uint64 target_id = 1;
}


message VoidResponse {

}
